### ISSUE TITLE
L064: Set preferred quotes for templated variables

### ISSUE THREAD

**Issue by dmohns** [sqlfluff/sqlfluff#3294]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

Currently, L064 ignores all STRING literals that involve quotes. For example a common Airflow templated STRING

```sql
SELECT '{{ ds }}'
```

is excluded from the the rule. (Assuming preferred quoting style is set to double quotes in this example). 

Now, as a user I could disable `ignore_templated_areas`. 
However, this is undesirable as it will pull in a bunch of false positives where the quotes are _inside_ the template. For example:

```sql
SELECT
    "a_string" AS col1,
    "{{ 'a_templated_string' }}" AS col2,
    "strings" IN {{ ('a', 'set', 'of', 'strings') }} AS col3
```

where the entire STRING is coming from a template.

```
L:   4 | P:  18 | L064 | Inconsistent use of preferred quote style 'double
                       | quotes'. Use "a" instead of 'a'.
L:   4 | P:  18 | L064 | Inconsistent use of preferred quote style 'double
                       | quotes'. Use "set" instead of 'set'.
L:   4 | P:  18 | L064 | Inconsistent use of preferred quote style 'double
                       | quotes'. Use "of" instead of 'of'.
L:   4 | P:  18 | L064 | Inconsistent use of preferred quote style 'double
                       | quotes'. Use "strings" instead of 'strings'.
```

It would be great, if we could expand L064 to cover cases where the quotes are outside the template. 

I would be willing to work on this if someone could point me into a loose direction on how this would be achievable.

### Use case

Increase consistency in SQL files w.r.t. to quoting styles.

### Dialect

Dialects with double quotes

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by barrywhart** [comment]
@dmohns: For the linting aspect, I think it will be similar to [L046](https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/rules/L046.py). In particular, note how it gets and loops over `templated_raw_slices`. For this issue,  you'll want to filter for slice type `literal` rather than `templated` at the beginning and end of the quoted string segment.

I don't think we have any rules yet which **fix** linting issues with templated code. It's probably doable, but I'm not sure about that, or exactly how it might be done.

For now, let's focus on linting, then once that is working, we can take a closer look at fixing.

**Comment by dmohns** [comment]
Great, thanks for the pointer @barrywhart 👍 

I agree focusing on linting first is a good idea. I'm not even sure if we want auto fix for this at all, as I can imagine some template setups might be relatively complex and we don't want to mess with it as part of the auto fix. I'll have a look at it in the up coming days.
